addons/stock_landed_costs/stock_landed_costs.py

```diff
@@ -286,7 +286,8 @@ def compute_landed_cost(self, cr, uid, ids, context=None):
 
                         if digits:
                             value = float_round(value, precision_digits=digits[1], rounding_method='UP')
-                            value = min(value, line.price_unit - value_split)
+                            fnc = min if line.price_unit > 0 else max
+                            value = fnc(value, line.price_unit - value_split)
```

Comment:
Hi, Josse,

The think is that if min is left, when landed cost is negative,
and comparison is made in negative values, the minimum amount
is the one with the largest absolute value,

That is ok only if only positive values were input.

for me the safety measure is great.

Best Regards.

Quien suscribe,

*Hbto*


On Thu, Oct 1, 2015 at 9:42 AM, Josse Colpaert <notifications@github.com>
wrote:

> In addons/stock_landed_costs/stock_landed_costs.py
> <https://github.com/odoo/odoo/pull/8714#discussion_r40917107>:
>
> > @@ -286,7 +286,8 @@ def compute_landed_cost(self, cr, uid, ids, context=None):
> >
> >                          if digits:
> >                              value = float_round(value, precision_digits=digits[1], rounding_method='UP')
> > -                            value = min(value, line.price_unit - value_split)
> > +                            fnc = min if line.price_unit > 0 else max
> > +                            value = fnc(value, line.price_unit - value_split)
>
> Ok for me, although I don't remember why this min/max is that necessary.
> It is more like a safety feature.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/odoo/odoo/pull/8714/files#r40917107>.
>